diff --git a/src/gx_head/engine/gx_pitch_tracker.cpp b/src/gx_head/engine/gx_pitch_tracker.cpp
index 3e3bd8a4..f5a565e8 100644
--- a/src/gx_head/engine/gx_pitch_tracker.cpp
+++ b/src/gx_head/engine/gx_pitch_tracker.cpp
@@ -175,6 +175,7 @@ void PitchTracker::start_thread(int priority, int policy) {
 		"PitchTracker",
 		_("error creating realtime thread - tuner not started"));
 	}
+        m_pthr = 0;
     }
     pthread_attr_destroy(&attr);
 }
